server/authz/rbac.go

```diff
@@ -153,3 +216,98 @@ func extractObject(c *gin.Context) string {
 	}
 	return emptyString
 }
+
+// getRbacProperty is used to read the rbacPropertiesPath file path and extract the policy provided as argument,
+func getRbacProperty(property string, config *viper.Viper) interface{} {
+	val := config.Get(property)
+	if val == nil {
+		return emptyString
+	}
+	return val
+}
+
+// getRBACScopes returns the scopes from the rbac properties file. If no scopes are provided, it returns Group as the
+// default scope. The scopes are used to determine the user identity token to be used for authorization.
+// If the scope is group, the user identity token will be the groups assigned to the user from the authentication
+// system. If the scope is email, the user identity token will be the email assigned to
+// the user from the authentication.
+// The scopes are provided as a comma separated list in the rbac properties file.
+// Example: policy.scopes=groups,email
+func getRBACScopes(config *viper.Viper) []string {
+	scopes := getRbacProperty(RbacPropertyScopes, config)
+	var retList []string
+	// If no scopes are provided, set Group as the default scope.
+	if scopes == emptyString {
+		retList = append(retList, ScopeGroup)
+		return retList
+	}
+	scopes = strings.Split(scopes.(string), ",")
+	for _, scope := range scopes.([]string) {
+		scope = strings.TrimSpace(scope)
+		retList = append(retList, scope)
+	}
+	return retList
+}
+
+// enforceCheck checks if the user has permission based on the Casbin model and policy.
+// It returns true if the user is authorized, otherwise false.
+func enforceCheck(enforcer *casbin.Enforcer, user, resource, object, action string) bool {
+	ok, _ := enforcer.Enforce(user, resource, object, action)
+	return ok
+}
+
+// ConfigFileReload is used to reload the config file when it is changed. This is used to reload the policy without
+// restarting the server. The config file is in the format of yaml. The config file is read by viper.
+func ConfigFileReload(e fsnotify.Event) {
+	logger.Infow("RBAC conf file updated:", "fileName", e.Name)
+	err := configReader.ReadInConfig()
+	if err != nil {
+		return
+	}
+	// update the scopes
+	newScopes := getRBACScopes(configReader)
+	currentScopes = newScopes
+	// update the default policy
+	policyDefault = getDefaultPolicy(configReader)
+	// clear the userPermCount cache
+	userPermCount = make(map[string]int)
+	logger.Infow("Auth Scopes Updated", "scopes", currentScopes)
+}
+
+// getDefaultPolicy returns the default policy from the rbac properties file. The default policy is used when the
+// requested resource is not present in the policy.
+// The default policy is provided in the rbac properties file in the format "policy.default: value"
+// Example: policy.default: deny
+func getDefaultPolicy(config *viper.Viper) string {
+	defaultPolicy := getRbacProperty(RbacPropertyDefaultPolicy, config)
+	if defaultPolicy == emptyString {
+		return emptyString
+	}
+	return defaultPolicy.(string)
```

Comment:
would it work?

```suggestion
	return getRbacProperty(RbacPropertyDefaultPolicy, config)
```

Reply:
We return an interface from the `getRbacProperty `, would want to cast that to string before sending. 
Or can it be done without that?

Reply:
sorry I meant would `return getRbacProperty(RbacPropertyDefaultPolicy, config).(string)` work? meaning if it's empty string, it also fits into this single statement.

Reply:
I have removed the extra check for emptyString